langchain/src/agents/tests/structured_output_runnables.int.test.ts

```diff
@@ -0,0 +1,118 @@
+import { zodToJsonSchema } from "zod-to-json-schema";
+import fs from "fs";
+import { z } from "zod";
+import {
+  AIMessage,
+  AgentAction,
+  AgentFinish,
+  AgentStep,
+} from "../../schema/index.js";
+import { RunnableSequence } from "../../schema/runnable/base.js";
+import { ChatPromptTemplate, MessagesPlaceholder } from "../../prompts/chat.js";
+import { ChatOpenAI } from "../../chat_models/openai.js";
+import { createRetrieverTool } from "../toolkits/index.js";
+import { RecursiveCharacterTextSplitter } from "../../text_splitter.js";
+import { HNSWLib } from "../../vectorstores/hnswlib.js";
+import { OpenAIEmbeddings } from "../../embeddings/openai.js";
+import { formatToOpenAIFunction } from "../../tools/convert_to_openai.js";
+import { AgentExecutor } from "../executor.js";
+import { formatForOpenAIFunctions } from "../format_scratchpad.js";
+
+/** Define a custom structured output parser. */
+const structuredOutputParser = (
```

Comment:
This is basically the equivalent of this right?

https://python.langchain.com/docs/modules/agents/how_to/agent_structured#create-the-custom-parsing-logic

Reply:
Yeah, just wrote that but in ts